manifests/pki.pp

```diff
@@ -1,51 +1,59 @@
+#   * If true, do *not* include SIMP's pki module, but still use pki::copy
+#     to manage certs in /etc/pki/simp_apps/simp_elasticsearch/x509
+#   * If false, do not include SIMP's pki module and do not use pki::copy
+#     to manage certs.  You will need to appropriately assign a subset of:
+#     * app_pki_dir
+#     * app_pki_key
+#     * app_pki_cert
+#     * app_pki_ca
+#     * app_pki_ca_dir
+#
+# @param app_pki_external_source
+#   * If pki = 'simp' or true, this is the directory from which certs will be
+#     copied, via pki::copy.  Defaults to /etc/pki/simp/x509.
+#
+#   * If pki = false, this variable has no effect.
+#
+# @param app_pki_dir
+#   This variable controls the basepath of $app_pki_key, $app_pki_cert,
+#   $app_pki_ca, $app_pki_ca_dir, and $app_pki_crl.
+#   It defaults to /etc/pki/simp_apps/simp_elasticsearch/x509.
+#
+# @param app_pki_key
+#   Path and name of the private SSL key file
+#
+# @param app_pki_cert
+#   Path and name of the public SSL certificate
+#
+# @param app_pki_ca_dir
+#   Path to the CA.
 #
 class simp_elasticsearch::pki(
-  Variant[Boolean,Enum['simp']] $pki               = simplib::lookup('simplib_options::pki' , { 'default_value' => false }),
-  Stdlib::AbsolutePath          $app_pki_cert      = "/etc/pki/simp/public/${facts['fqdn']}.pub",
-  Stdlib::AbsolutePath          $app_pki_key       = "/etc/pki/simp/private/${facts['fqdn']}.pem",
-  Stdlib::AbsolutePath          $app_pki_ca_dir    = '/etc/pki/simp/cacerts',
-  Stdlib::AbsolutePath          $app_pki_dir       = '/etc/pki/elasticsearch',
-  Stdlib::AbsolutePath          $app_pki_external_source = simplib::lookup('simplib_options::app_pki_external_source',{ 'default_value' => '/etc/pki/simp'}),
-  String                        $group             = 'apache',
-  String                        $owner             = 'root'
+  Variant[Boolean,Enum['simp']] $pki                     = simplib::lookup('simp_options::pki' , { 'default_value' => false }),
+  Stdlib::Absolutepath          $app_pki_external_source = simplib::lookup('simp_options::pki::source', { 'default_value' => '/etc/pki/simp/x509' }),
+  Stdlib::AbsolutePath          $app_pki_dir             = '/etc/pki/simp_apps/simp_elasticsearch/x509',
+  Stdlib::AbsolutePath          $app_pki_cert            = "${app_pki_dir}/public/${facts['fqdn']}.pub",
+  Stdlib::AbsolutePath          $app_pki_key             = "${app_pki_dir}/private/${facts['fqdn']}.pem",
+  Stdlib::AbsolutePath          $app_pki_ca_dir          = "${app_pki_dir}/cacerts",
+  String                        $group                   = 'apache',
+  String                        $owner                   = 'root'
 ){
 
   if $pki {
-
-    ::pki::copy { $app_pki_dir :
+    ::pki::copy { 'simp_elasticsearch' :
       source => $app_pki_external_source,
       pki    => $pki,
       owner  => $owner,
       group  => $group,
       notify => $::simp_elasticsearch::http_service_resource
```

Comment:
Even if `$::simp_elasticsearch::http_service_resource` was a valid variable, it isn't guaranteed to be available since `simp_elasticsearch` might not have been included.

I don't think there is a use case where it is desirable to include `simp_elasticsearch::pki` _without_ `simp_elasticsearch`, so `simp_elasticsearch::pki` might be better off as a private class with its data supplied directly by `simp_elasticsearch`.  That why the standard [SIMP module layout](https://simp-project.atlassian.net/wiki/display/SD/The+New+Layout+for+all+SIMP+Modules) handles pki that way it does:
- a  "front controller" pattern in init.pp: https://github.com/simp/puppet-module-skeleton/blob/master/skeleton/manifests/init.pp.erb#L72-L76
- an `asset_private()` in config/pki.pp:  https://github.com/simp/puppet-module-skeleton/blob/master/skeleton/manifests/config/pki.pp.erb#L7

(PS: :+1: to our policy of testing with `STRICT_VARIABLES=yes` for catching this )